logstash-core/spec/logstash/instrument/wrapped_write_client_spec.rb

```diff
@@ -18,39 +18,54 @@
 
   subject { described_class.new(write_client, pipeline, metric, plugin) }
 
+  def threaded_read_client(read_client)
+    Thread.new(read_client) do |_read_client|
+      started_at = Time.now
+
+      batch_size = 0
+      loop {
+        if Time.now - started_at > 60
```

Comment:
cool, we could reuse this pattern in many places, even maybe add this feature to `stud/try`